src/main/resources/edu/cornell/kfs/coa/businessobject/datadictionary/ObjectCode.xml

```diff
@@ -1,356 +1,499 @@
+          parent="ObjectCode-financialObjectCode"
+          p:label="Historical Financial Object Code"
+          p:name="historicalFinancialObjectCode"
+          p:required="false"
+          p:shortLabel="HistFinObjCde"/>
+
+    <bean id="ObjectCode-active"
+          parent="ObjectCode-active-parentBean"/>
+    <bean id="ObjectCode-active-parentBean"
+          abstract="true"
+          parent="GenericAttributes-activeIndicator"
+          p:name="active"/>
+
+    <bean id="ObjectCode-financialBudgetAggregationCd"
+          parent="ObjectCode-financialBudgetAggregationCd-parentBean"/>
+    <bean id="ObjectCode-financialBudgetAggregationCd-parentBean"
+          abstract="true"
+          parent="BudgetAggregationCode-code"
+          p:name="financialBudgetAggregationCd"
+          p:required="true"/>
+
+    <bean id="ObjectCode-finObjMandatoryTrnfrelimCd-parentBean"
+          abstract="true"
+          parent="MandatoryTransferEliminationCode-code"
+          p:name="finObjMandatoryTrnfrelimCd">
+        <property name="control">
+            <bean parent="SelectControlDefinition"
+                  p:includeKeyInLabel="false"
+                  p:valuesFinder-ref="mandatoryTransferEliminationCodeValuesFinder"/>
+        </property>
+    </bean>
+
+    <bean id="ObjectCode-financialFederalFundedCode"
+          parent="ObjectCode-financialFederalFundedCode-parentBean"/>
+    <bean id="ObjectCode-financialFederalFundedCode-parentBean"
+          abstract="true"
+          parent="FederalFundedCode-code"
+          p:name="financialFederalFundedCode"
+          p:required="true">
+        <property name="control">
+            <bean parent="SelectControlDefinition"
+                  p:includeKeyInLabel="false"
+                  p:valuesFinder-ref="federalFundedCodeValuesFinder"/>
+        </property>
+    </bean>
+
+    <bean id="ObjectCode-nextYearFinancialObjectCode"
+          parent="ObjectCode-nextYearFinancialObjectCode-parentBean"/>
+    <bean id="ObjectCode-nextYearFinancialObjectCode-parentBean"
+          abstract="true"
+          parent="ObjectCode-financialObjectCode"
+          p:label="Next Year Object Code"
+          p:name="nextYearFinancialObjectCode"
+          p:required="false"
+          p:shortLabel="NextYearObjCode"/>
+
+    <bean id="ObjectCode-financialObjectLevel.financialConsolidationObjectCode"
+          parent="ObjectCode-financialObjectLevel.financialConsolidationObjectCode-parentBean"/>
+    <bean id="ObjectCode-financialObjectLevel.financialConsolidationObjectCode-parentBean"
+          abstract="true"
+          parent="ObjectConsolidation-finConsolidationObjectCode"
+          p:name="financialObjectLevel.financialConsolidationObjectCode"/>
+
+    <bean id="ObjectCode-extension-sunyObjectCode" parent="ObjectCode-extension-sunyObjectCode-parentBean"/>
+    <bean id="ObjectCode-extension-sunyObjectCode-parentBean" abstract="true" parent="ObjectCodeExtendedAttribute-sunyObjectCode"
```

Comment:
you might want to consider formatting these definitions for the extension properties similar to the ones above